src/components/PeopleTable.tsx

```diff
@@ -1,5 +1,72 @@
+import { Link, useParams, useLocation } from 'react-router-dom';
+
+import classNames from 'classnames';
+
+import { Person } from '../types';
+import { SearchLink } from './SearchLink';
+
 /* eslint-disable jsx-a11y/control-has-associated-label */
-export const PeopleTable = () => {
+type Props = {
+  people: Person[];
+};
+
+type PersonLinkProps = {
+  person?: Person;
+  name?: string | null;
+};
+
+export const PeopleTable: React.FC<Props> = ({ people }) => {
+  const { peopleSlug } = useParams();
+  const location = useLocation();
+  const searchParams = new URLSearchParams(location.search);
+
+  const PersonLink = ({ person, name }: PersonLinkProps) => {
+    const slug = person?.slug;
+    const linkName = person?.name || name || '-';
+    const parentsName = people.find(p => p.name === linkName);
+    const isWoman = person?.sex === 'f' || parentsName?.sex === 'f';
+
+    if (slug || parentsName) {
+      return (
+        <td>
+          <Link
+            to={`/people/${slug || parentsName?.slug}`}
+            className={classNames({
+              'has-text-danger': isWoman,
+            })}
+          >
+            {linkName}
+          </Link>
+        </td>
+      );
+    }
+
+    return <td>{linkName}</td>;
+  };
```

Comment:
It should be a separate component in this case. move it outside PeopleTable component